Libraries/LibWeb/Bindings/MainThreadVM.cpp

```diff
@@ -677,9 +678,11 @@ void queue_mutation_observer_microtask(DOM::Document const& document)
         for (auto& observer : surrounding_agent.mutation_observers)
             notify_set.append(&observer);
 
-        // FIXME: 3. Let signalSet be a clone of the surrounding agent’s signal slots.
+        // 3. Let signalSet be a clone of the surrounding agent’s signal slots.
+        auto signal_set = surrounding_agent.signal_slots;
 
-        // FIXME: 4. Empty the surrounding agent’s signal slots.
+        // 4. Empty the surrounding agent’s signal slots.
+        surrounding_agent.signal_slots.clear();
```

Comment:
If we're just cloning then clearing, we can avoid potentially expensive copies here:
```c++
// 3. Let signalSet be a clone of the surrounding agent’s signal slots.
// 4. Empty the surrounding agent’s signal slots.
auto signal_set = move(surrounding_agent.signal_slots);
```